sopel/test_tools.py

```diff
@@ -263,6 +260,7 @@ def insert_into_module(func, module_name, base_name, prefix):
     setattr(module, func.__name__, func)
 
 
+@tools.deprecated('py.test run tests from @example directly', '7.1', '8.0')
```

Comment:
I think this is what you meant (but it's already been the case for a _long_ time, so we probably should've deprecated this sooner 😅)

```suggestion
@tools.deprecated('pytest now runs @plugin.example tests directly', '7.1', '8.0')
```

I want to mention `@module.example` too somehow, but can't think of a way that doesn't make the deprecation message extra long and cluttered. :/